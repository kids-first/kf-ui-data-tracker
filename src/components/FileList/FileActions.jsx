import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {Mutation} from 'react-apollo';
import {GET_STUDY_BY_ID} from '../../state/queries';
import {DELETE_FILE, FILE_DOWNLOAD_URL} from '../../state/mutations';
import {KF_STUDY_API} from '../../common/globals';
import {Icon} from 'kf-uikit';

const FileActions = ({node, studyId, className}) => {
  const FileActionsClass = classes('FileActions', className);

  const FileActionButton = classes('FileActionButton');
  if (node) {
    return (
      <Mutation mutation={FILE_DOWNLOAD_URL} key={node.kfId}>
        {downloadFile => (
          <Mutation
            mutation={DELETE_FILE}
            update={(cache, {data: {deleteFile}}) => {
              // Re-writes the study in the cache with the deleted file
              // removed.
              const {studyByKfId} = cache.readQuery({
                query: GET_STUDY_BY_ID,
                variables: {kfId: studyId},
              });
              const deleteIndex = studyByKfId.files.edges.findIndex(
                edge => edge.node.kfId === deleteFile.kfId,
              );
              if (deleteIndex < 0) {
                return studyByKfId;
              }
              studyByKfId.files.edges.splice(deleteIndex, 1);
              const data = {
                studyByKfId: {
                  ...studyByKfId,
                  files: {
                    __typename: 'FileNodeConnection',
                    edges: studyByKfId.files.edges,
                  },
                },
              };
              cache.writeQuery({
                query: GET_STUDY_BY_ID,
                data,
              });
            }}
            key={node.kfId}
          >
            {(deleteFile, {loading, error}) => (
              <FileActionButtons
                {...{
                  node,
                  studyId,
                  deleteFile,
                  downloadFile,
                  loading,
                  className,
                  error,
                }}
              />
            )}
          </Mutation>
        )}
      </Mutation>
    );
  }
  return null;
};

export default FileActions;
