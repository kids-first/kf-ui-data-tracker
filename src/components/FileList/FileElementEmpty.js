import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import Badge from '../Badge/Badge';
/**
 * Displays unordered studies in grid view (include empty stage message)
 */

const FileElementEmpty = ({className}) => {
  const fileElementEmptyClass = classes(
    'FileList--Element',
    'sm:flex-no-wrap',
    className,
  );

  return (
    <li className={fileElementEmptyClass}>
      <div className="FileList--ElementBadge sm:w-48">
        <Badge state="empty" />
        <Badge state="empty" />
      </div>
      <div className="flex-initial">
        <p className="mt-0 pt-2 font-bold text-sm">
          <span className="bg-lightGrey px-12 mx-4" />
          <span className="bg-lightGrey px-8 mx-4" />
          <span className="bg-lightGrey px-12 mx-4" />
        </p>
        <p className="mt-0 font-normal text-grey text-xs">
          <span className="bg-lightGrey px-12 mx-4" />
          <span className="bg-lightGrey px-8 mx-4" />
          <span className="bg-lightGrey px-6 mx-4" />
        </p>
      </div>
    </li>
  );
};

FileElementEmpty.propTypes = {
  /** Any additional classes to be applied to the empty study element*/
  className: PropTypes.string,
};

FileElementEmpty.defaultProps = {
  className: null,
};

export default FileElementEmpty;
