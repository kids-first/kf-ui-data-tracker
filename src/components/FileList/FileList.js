import React, {useState, Fragment} from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import FileElement from './FileElement';
import ReactPaginate from 'react-paginate';
import NotificationBar from '../NotificationBar/NotificationBar';
import {fileSortedVersions} from '../../common/fileUtils';
/**
 * Displays list of study files
 */
const FileList = ({className, fileList, studyId}) => {
  let fileListClass = classes('FileList', className);
  const perPage = 5;
  const [page, setPage] = useState(0);
  const handlePageClick = data => {
    setPage(data.selected);
  };
  const pageCount = () => {
    if (Math.ceil(fileList.length / perPage) < page + 1) {
      setPage(page - 1);
    }
    return Math.ceil(fileList.length / perPage);
  };
  const offset = fileList.slice(perPage * page, perPage * page + perPage);

  const showNotification = () => {
    var statusList = [];
    var i;
    for (i in fileList) {
      const sortedVersions = fileSortedVersions(fileList[i].node);
      const versionStatus =
        sortedVersions.length > 0 ? sortedVersions[0].node.state : null;
      statusList.push(versionStatus);
    }
    if (statusList.includes('CHN')) {
      return (
        <NotificationBar
          borderColor="rgba(221,31,42,1)"
          backgroundColor="rgba(221,31,42,0.1)"
          icon="error"
          message="Changes Needed"
          note="The DRC has requested that you make changes to some of your documents."
        />
      );
    } else {
      return null;
    }
  };

  return (
    <Fragment>
      {showNotification()}
      <ul className={fileListClass}>
        {fileList.length ? (
          offset.map(({node}) => (
            <FileElement key={node.kfId} fileListId={studyId} fileNode={node} />
          ))
        ) : (
          <h3 className="FileList--Empty">You don't have any files yet.</h3>
        )}
      </ul>
      {pageCount() > 0 && (
        <Fragment>
          <p className="m-0 text-sm text-mediumGrey text-right">
            {offset.length} of {fileList.length} files
          </p>
          <ReactPaginate
            previousLabel={'<'}
            nextLabel={'>'}
            breakLabel={'...'}
            breakClassName={'break-me'}
            pageCount={pageCount()}
            marginPagesDisplayed={2}
            pageRangeDisplayed={3}
            forcePage={page}
            onPageChange={handlePageClick}
            containerClassName={'Pagination'}
            subContainerClassName={'Pagination'}
            activeClassName={'Pagination--active'}
          />
        </Fragment>
      )}
    </Fragment>
  );
};

FileList.propTypes = {
  /** Any additional classes to be applied to the study list*/
  className: PropTypes.string,
  /** Array of study object*/
  fileList: PropTypes.array,
  /** Kids First unique study identifier (SD_XXXXXXXX) */
  studyId: PropTypes.string.isRequired,
};

FileList.defaultProps = {
  className: null,
  fileList: [],
  studyId: null,
};

export default FileList;
