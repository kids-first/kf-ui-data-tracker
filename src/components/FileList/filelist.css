/* FileList and FileListElement CSS */
.FileList {
  @extend .list-reset,
    .min-w-full,
    .my-20,
    .p-0,
    .ml-0,
    .border-t,
    .border-grey;
}

.FileList--Element {
  @extend .list-reset, .w-full, .py-8, .border-b, .border-grey, .text-black;
  transition: all 0.25s ease-in;
}
.FileList--Element-title {
  transition: all 0.25s ease-in;
  color: black;
}

.FileList--Element:hover,
.FileList--Element:focus {
  @extend .bg-lightGrey, .shadow, .border-grey-lightest;
  .FileList--Element-desc {
    height: 25px;
  }
  .FileActions {
    opacity: 1;
  }
}

.FileList--Element:hover .FileList--Element-name,
.FileList--Element:focus .FileList--Element-name {
  color: #2b388f;
  transition: all 0.5s ease;
}

.FileList--Element-desc {
  height: 0;
  transition: height 0.5s ease;
  overflow: hidden;
}

.FileList--Empty {
  @extend .text-grey, .border-b, .border-grey, .m-0, .py-16, .px-32;
}

.FileList--ElementBadge {
  @extend .row-2, .cell-12, .flex-initial, .flex, .flex-row, .flex-wrap;
}

.FileList--ElementButton {
  @extend .inline-block, .mr-8, .p-0;
}

.FileStatus {
  @extend .mr-16, .font-normal, .text-xs;
}

.FileStatus--loading {
  @extend .mr-16, .px-16, .bg-lightGrey, .text-lightGrey;
}

.Pagination {
  @extend .min-w-full, .flex, .list-reset, .justify-center;
}

.Pagination li {
  @extend .border, .border-lightGrey, .text-lightBlue;
}

.Pagination li a {
  @extend .block, .py-8, .px-16;
}

.Pagination--active {
  @extend .bg-lightBlue;
}

.Pagination--active a {
  @extend .text-white;
}
