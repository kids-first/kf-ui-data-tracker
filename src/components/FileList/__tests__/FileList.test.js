import React from 'react';
import wait from 'waait';
import {MemoryRouter} from 'react-router-dom';
import {MockedProvider} from 'react-apollo/test-utils';
import {render, cleanup, fireEvent} from 'react-testing-library';
import FileList from '../FileList';
import studyByKfId from './studyByKfId';

afterEach(cleanup);

it('renders with files', async () => {
  const files = studyByKfId.data.studyByKfId.files.edges;

  const tree = render(
    <MemoryRouter>
      <MockedProvider>
        <FileList fileList={files} studyId={'SD_00000000'} />
      </MockedProvider>
    </MemoryRouter>,
  );

  // Sort files by create date by clicking on the file sorting dropdown
  const sortDropdown = tree.getByText(/Date option/i);
  fireEvent.click(sortDropdown);
  await wait();
  const sortByCreateDate = tree.getByText(/Modified date/i);
  fireEvent.click(sortByCreateDate);
  await wait();

  expect(tree.container).toMatchSnapshot();

  // 12 files in total, page 1 contains 10
  const lisPage1 = tree.queryAllByTestId('file-item');
  expect(lisPage1.length).toBe(10);

  // Click on the next button to go to next page

  let button = tree.getByLabelText('Next page');
  button.click();

  await wait();

  // 12 files in total, 10 files per page, page 2 should have 2 files
  const lisPage2 =  tree.queryAllByTestId('file-item');
  expect(lisPage2.length).toBe(2);
});

it('renders without files', () => {
  const tree = render(
    <MemoryRouter>
      <FileList fileList={[]} studyId={'SD_00000000'} />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();

  const lis = tree.queryAllByTestId('file-item');
  expect(lis.length).toBe(0);

  // Hide pagination when no files exist
  const pagination = tree.container.querySelectorAll('.Pagination');
  expect(pagination.length).toBe(0);
});
