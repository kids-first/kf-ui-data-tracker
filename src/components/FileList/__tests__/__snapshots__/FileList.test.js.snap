// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with files 1`] = `
<div>
  <div
    class="ui basic compact left floated segment noMargin noHorizontalPadding"
  >
    <span
      class="smallLabel"
    >
      Filter by:
    </span>
    <div
      aria-expanded="false"
      class="ui selection dropdown"
      role="listbox"
      tabindex="0"
    >
      <div
        aria-live="polite"
        class="default text"
        role="alert"
      >
        Approval status
      </div>
      <i
        aria-hidden="true"
        class="dropdown icon"
      />
      <div
        class="menu transition"
      >
        <div
          aria-checked="false"
          aria-selected="true"
          class="selected item"
          role="option"
          style="pointer-events: all;"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <div
            class="ui teal tiny basic label"
          >
            Approved
          </div>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <div
            class="ui red tiny basic label"
          >
            Changes needed
          </div>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <div
            class="ui blue tiny basic label"
          >
            Processed
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ui basic compact left floated segment noMargin noHorizontalPadding"
  >
    <div
      aria-expanded="false"
      class="ui selection dropdown"
      role="listbox"
      tabindex="0"
    >
      <div
        aria-live="polite"
        class="default text"
        role="alert"
      >
        File type
      </div>
      <i
        aria-hidden="true"
        class="dropdown icon"
      />
      <div
        class="menu transition"
      >
        <div
          aria-checked="false"
          aria-selected="true"
          class="selected item"
          role="option"
          style="pointer-events: all;"
        >
          <small
            class="text"
          >
            <i
              aria-hidden="true"
              class="shipping icon"
            />
             Shipping Manifest
          </small>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <small
            class="text"
          >
            <i
              aria-hidden="true"
              class="hospital icon"
            />
             Clinical/Phenotype Data
          </small>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <small
            class="text"
          >
            <i
              aria-hidden="true"
              class="dna icon"
            />
             Sequencing Manifest
          </small>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <small
            class="text"
          >
            <i
              aria-hidden="true"
              class="question icon"
            />
             Other
          </small>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ui basic compact left floated segment noMargin noHorizontalPadding"
  >
    <span
      class="smallLabel"
    >
      Sorted by:
    </span>
    <div
      aria-expanded="false"
      class="ui selection dropdown"
      role="listbox"
      tabindex="0"
    >
      <div
        aria-live="polite"
        class="text"
        role="alert"
      >
        Modified date
      </div>
      <i
        aria-hidden="true"
        class="dropdown icon clear"
      />
      <div
        class="menu transition"
      >
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            Create date
          </span>
        </div>
        <div
          aria-checked="true"
          aria-selected="true"
          class="active selected item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            Modified date
          </span>
        </div>
      </div>
    </div>
    <button
      class="ui basic icon button"
    >
      <i
        aria-hidden="true"
        class="sort content ascending icon"
      />
    </button>
  </div>
  <div
    class="ui basic compact right floated segment noMargin noHorizontalPadding"
  >
    <div
      class="ui icon input"
    >
      <input
        type="text"
        value=""
      />
      <i
        aria-hidden="true"
        class="search icon"
      />
    </div>
  </div>
  <table
    class="ui selectable stackable very basic very compact table"
  >
    <tbody
      class=""
    >
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            pattern.flac
          </span>
          <p
            class="noMargin"
          >
            Shoulder body answer especially large big risk its. Sort fly note field fish. Network him identify r...
          </p>
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_JT0GRZP7
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2017-11-06T13:13:56.000Z"
                  title="6 Nov 2017"
                >
                  1 year ago
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                6.6 kB
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="hospital icon"
              />
               Clinical/Phenotype Data
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            those.wav
          </span>
          <p
            class="noMargin"
          >
            Feel industry quality direction resource truth. Affect green yard various play character simply rema...
          </p>
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_DLXEQZV1
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2018-07-09T05:01:45.000Z"
                  title="9 Jul 2018"
                >
                  10 months ago
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                7.0 kB
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               unknown
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            force.webm
          </span>
          <p
            class="noMargin"
          >
            Safe always adult sometimes. Enter full agree still son expect write. According small hold someone.
          </p>
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_FPVBM0JF
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2018-09-19T08:33:31.000Z"
                  title="19 Sept 2018"
                >
                  7 months ago
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                440 B
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="hospital icon"
              />
               Clinical/Phenotype Data
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            hand.css
          </span>
          <p
            class="noMargin"
          >
            This democratic analysis car here likely. Protect speech exactly three order different wife.
          </p>
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_6JBSGVWE
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2018-10-31T05:37:11.000Z"
                  title="31 Oct 2018"
                >
                  6 months ago
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                9.6 kB
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               unknown
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            3.jpg
          </span>
          <p
            class="noMargin"
          />
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_SFA9WW62
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2019-05-20T14:34:21.331Z"
                  title="20 May 2019"
                >
                  4 weeks from now
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                75.5 kB
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            3.jpg
          </span>
          <p
            class="noMargin"
          />
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_SFA9WW01
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2019-05-20T14:34:21.331Z"
                  title="20 May 2019"
                >
                  4 weeks from now
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                75.5 kB
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            3.jpg
          </span>
          <p
            class="noMargin"
          />
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_SFA9WW02
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2019-05-20T14:34:21.331Z"
                  title="20 May 2019"
                >
                  4 weeks from now
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                75.5 kB
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            3.jpg
          </span>
          <p
            class="noMargin"
          />
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_SFA9WW03
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2019-05-20T14:34:21.331Z"
                  title="20 May 2019"
                >
                  4 weeks from now
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                75.5 kB
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            3.jpg
          </span>
          <p
            class="noMargin"
          />
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_SFA9WW04
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2019-05-20T14:34:21.331Z"
                  title="20 May 2019"
                >
                  4 weeks from now
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                75.5 kB
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
      >
        <td
          class="collapsing single line center aligned"
        >
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </td>
        <td
          class=""
        >
          <span
            class="ui medium header"
          >
            4.jpg
          </span>
          <p
            class="noMargin"
          />
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <button
                class="ui mini basic icon left labeled button"
              >
                <i
                  aria-hidden="true"
                  class="copy icon"
                />
                SF_AZMTRAND
              </button>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                Modified
                 
                <time
                  datetime="2019-05-20T14:38:20.880Z"
                  title="20 May 2019"
                >
                  4 weeks from now
                </time>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <div
                class="description"
              >
                61.0 kB
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </div>
          </div>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui mini vertical buttons"
          >
            <button
              class="ui basic icon button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </button>
            <button
              class="ui basic icon button"
              data-testid="delete-button"
            >
              <i
                aria-hidden="true"
                class="trash alternate icon"
              />
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="ui basic right aligned segment"
  >
    Showing 
    10
     of 
    12
     files
     
    <div
      aria-label="Pagination Navigation"
      class="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current="false"
        aria-disabled="false"
        aria-label="Previous page"
        class="item"
        tabindex="0"
        type="prevItem"
        value="1"
      >
        <i
          aria-hidden="true"
          class="chevron left icon"
        />
      </a>
      <a
        aria-current="true"
        aria-disabled="false"
        class="active item"
        tabindex="0"
        type="pageItem"
        value="1"
      >
        1
      </a>
      <a
        aria-current="false"
        aria-disabled="false"
        class="item"
        tabindex="0"
        type="pageItem"
        value="2"
      >
        2
      </a>
      <a
        aria-current="false"
        aria-disabled="false"
        aria-label="Next page"
        class="item"
        tabindex="0"
        type="nextItem"
        value="2"
      >
        <i
          aria-hidden="true"
          class="chevron right icon"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`renders without files 1`] = `
<div>
  <div
    class="ui basic segment"
  >
    <div
      class="ui icon center aligned header"
    >
      <i
        aria-hidden="true"
        class="file alternate outline icon"
      />
      You don't have any documents yet.
    </div>
  </div>
</div>
`;
